#169817886:user edit trip request

mujohn26 opened the pull request:
#### What does this PR do?

-  user should be able to edit trip request which are still open 

#### Description of Task to be completed?

-  user should provide JWT to be checked
-  user should pass the trip id in the header
-  user should send the editable request on that specific trip
#### How should this be manually tested?
-  clone it 
- run npm i to install dependencies
- run `npm run db-migrate` to create databases
- run `npm  run start`
-  use `pi/v1/trips/:tripId`

#### Any background context you want to provide?

- you must provide JWT token to able to perform the task
- you must be logged in and your account is verified
- you will be able to edit the only request which is pending or rejected 

#### What are the relevant pivotal tracker stories?

[#169817886](https://www.pivotaltracker.com/story/show/169817886)
#### Screenshots (if appropriate)

<img width="1133" alt="Screen Shot 2020-01-16 at 21 05 52" src="https://user-images.githubusercontent.com/52497006/72554786-0fdaa200-38a4-11ea-989a-9f2c36af5b22.png">









#### Questions:

N/A

Jajabenit250 commented on src/controllers/trip.controller.js:
```diff
@@ -39,27 +39,42 @@ class tripController {
+        req.body,
+        tripid,
+        type
+      );
```
Unnecessary change. I think we might have the previous code we don't need to change it ?

Jajabenit250 commented on src/controllers/trip.controller.js:
```diff
@@ -39,27 +39,42 @@ class tripController {
+        tripId,
+        userId,
+        managerId,
+        status
```
Unnecessary change. I think we might have the previous code we don't need to change it ?

mujohn26 replied:
Thanks for the feedback benit, surely I have not made any change about the code and the functionality the reason behind, it's because I used prettier formater to have a good design, on other side no changes I made.

Jajabenit250 replied:
@mujohn26 Kindly check if it is possible to remain with old code that you are not changing. because unnecessary changes will cause merge conflict in future.

Jajabenit250 commented on src/controllers/trip.controller.js:
```diff
@@ -39,27 +39,42 @@ class tripController {
+        tripRequest.id,
+        'http://localhost:3000/api/v1/notifications'
+      );
+      return response.successMessage(
```
Unnecessary change. I think we might have the previous code we don't need to change it ?

Jajabenit250 reviewed (commented):
Good progress so far but kindly attend to my feedback. the general feedback is that you are making many changes and it make reviewer to not know what you actually did.

bdushimi commented on src/middlewares/accomodation.middleware.js:
```diff
@@ -1,6 +1,7 @@
-import tripServicres from '../services/trip.services';
-import db from '../database/models';
+// import tripServicres from '../services/trip.services';
+// import db from '../database/models';
```
@mujohn26 Kindly always remove commented lines.

bdushimi commented on src/controllers/trip.controller.js:
```diff
@@ -56,7 +56,7 @@ class TripController {
       };
       const tripRequest = await tripService.CreateTripRequest(requestData);
       await NotificationService.sendNotification('trip_request_event', managerId, `New ${type} trip request.`, `${req.user.firstName} has requested a new ${type}`, tripRequest.id, 'http://localhost:3000/api/v1/notifications');
```
@mujohn26 May you update this URL (http://localhost:3000/api/v1/notifications')) to point to a dynamic URL, that uses the environment variable instead of hardcoding it?

bdushimi commented on src/controllers/trip.controller.js:
```diff
@@ -221,5 +222,82 @@ class TripController {
+   * @param {*} res response
+   * @returns {Object} return user response
+   */
+  static async combineUpdateTripRequestConctroller(req, res) {
```
@mujohn26 Kindly update the name of this function. Its description and its name are not matching.

Jajabenit250 replied:
what do you mean by combineUpdateTripRequestController i would like if you kindly check the name twice and rename the function according to what it is doing. and since it is a function inside class called TripController i think there no need of calling it controller.

bdushimi commented on src/services/Queries.js:
```diff
@@ -296,5 +317,97 @@ class Queries {
+    try {
+      const updatedRequest = await table.update(
+
+
```
@mujohn26 Kindly remove these white spaces

bdushimi commented on src/controllers/trip.controller.js:
```diff
@@ -221,5 +222,82 @@ class TripController {
+
+
+      await tripRequestService.updateTripRequest(tripId, req.body);
+      await tripRequestService.updateRequestStatus(tripId);
```
@mujohn26 What's the difference between these two functions (at 253 & 254 lines)? Aren't they doing same thing?

mujohn26 replied:
I changed them for a better description,one is updating the trip and another one is updating the trip request status.

bdushimi commented on src/routes/trip.route.js:
```diff
@@ -151,32 +123,44 @@ router.get('/my-trip-requests', verifyToken.headerToken, verifyUser, tripControl
+  TripMiddleware.checkLocations,
+  AccomodationMiddleware.findAccommodationByCity,
+  TripMiddleware.multiCityDataValidation,
+  tripController.combineUpdateTripRequestConctroller
```
@mujohn26 A couple of things here :
1. No need to add "Controller" to the name of this function because it is already part of the controller class.
2. This function seems to be updating the tripRequest yet the route is about updating a trip information.

bdushimi commented on src/services/Queries.js:
```diff
@@ -296,5 +317,97 @@ class Queries {
+
+  /**
+   *
+   * This method will be user to delete the selected trip in multi city
```
@mujohn26 Kindly proofread your description and make sure they're grammatically correct, concise and are conveying the message easily

bdushimi commented on src/services/request.services.js:
```diff
@@ -24,6 +24,87 @@ class TripRequestService {
+   * @param {string} tripId the id of the selected trip
+   * @returns {object} updated data
+   */
+  static async updateRequestStatus(tripId) {
```
@mujohn26 Which request are you referring to here?

bdushimi commented on src/services/Queries.js:
```diff
@@ -296,5 +317,97 @@ class Queries {
+   * @param {Object} data it gets the data from req.body
+   * @returns {object} updated data
+   */
+  static async deleteMultiCityTripRequest(table, tripId) {
```
@mujohn26 is this function about deleting only a multi-city request or multi-city trip? I think there should be generic functions to delete trip-request and trip given IDs

Jajabenit250 commented on src/controllers/trip.controller.js:
```diff
@@ -221,5 +222,82 @@ class TripController {
+  }
+
+  /**
+   * This controller will help a user to edits multiple city request
```
user to edit instead of edits

Jajabenit250 commented on src/middlewares/trip.middleware.js:
```diff
@@ -99,5 +101,26 @@ class TripMiddleware {
   }
+
+  /**
+   * this function is used to check status of the trip so that it can edited
```
it can be edited.

Jajabenit250 reviewed (commented):
Good Work. But Kindly attend to change requested.

shemaeric1234 commented on src/services/request.services.js:
```diff
@@ -24,6 +24,87 @@ class TripRequestService {
+   * @returns { Object } response data
+   */
+  static async updateMultiCityTripRequest(body, tripId) {
+    const data = {
```
the same here  `updateMultiCityTripRequest`  and `updateTrip` methods are doing the same thing, so you can find a way of using a single method.

shemaeric1234 commented on src/services/Queries.js:
```diff
@@ -299,5 +320,95 @@ class Queries {
+   * @returns {object} updated data
+   */
+  static async updateRequestStatusById(table, id) {
+    try {
```
Hey, @mujohn26 you did a good job so far, however, there is something you can change.
`updateRequestStatusById`, `updateMultiCityTripRequest` and `updateTrip`  methods are doing the same thing, so you can find a way of using a single method, and think that can be better.

jo-rdan reviewed (approved):
Good work @mujohn26

mujohn26 committed:
feat(edit-request):edit request

- user should be able to edit trip request which is still open

[finish #169817894]
